src/store/courses/reducers.ts

```diff
@@ -8,9 +8,11 @@ import {
 const initialState: CourseStateType = { isCourse: true };
 
 export const courses = (
-  state: CourseStateType = initialState,
+  inputState: CourseStateType,
```

Comment:
pourquoi ce changement ?

Reply:
Ce changement est dû a une double contrainte : 
- la regle eslint : "default-param-last", qui impose que les parametres ayant un "par defaut" soit appelés en derniers.
- la syntaxe du reducer qui nous contraint de respecet l'ordre : `state`, puis `action`.

le pb est relevé ici : https://stackoverflow.com/questions/62259351/react-redux-error-default-parameters-should-be-last-default-param-last

J'ai vu plusieurs solutions a cela : 
- mettre un "par defaut" a l'argument `action` egalement. Mais que mettre ?
- initialiser state au debut du corps de la fct (ce que j'ai fait)
- desactiver la regle eslint

Tu vois une autre manere de faire ?

Reply:
tu pourrais mettre `action = {}` par defaut qu'en penses tu ?

Reply:
initialiser action avec {} fait planter ts dans `switch (action.type)` du coup j'ai plutot initialisé avec `{ type: null }`. T'en penses quoi ?

Reply:
tu pourrais mettre `{ type: '' }` qu'en penses tu ?